### PR TITLE
python311Packages.qcodes: fix src, wheel version

### LINKED ISSUE #288158: python311Packages.qcodes specifies version 0.0 in METADATA
### Describe the bug
```
$ nix-build -A python311.pkgs.qcodes
$ cat result/lib/python3.11/site-packages/qcodes-0.0.dist-info/METADATA
Metadata-Version: 2.1
Name: qcodes
Version: 0.0
Summary: Python-based data acquisition framework developed by the Copenhagen / Delft / Sydney / Microsoft quantum computing consortium
...
```
This breaks qcodes-contrib-drivers.

### Notify maintainers
@evilmav 

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by eclairevoyant**
## Description of changes

python311Packages.qcodes:
- fix repo owner
- fix wheel version

Fixes #288158.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/qcodes/default.nix [hunk 1]
```diff
@@ -147,6 +147,11 @@ buildPythonPackage rec {
     "qcodes"
   ];
 
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace-fail '"0.0"' '"${version}"'
```
Comment (a-n-n-a-l-e-e) [inline pkgs/development/python-modules/qcodes/default.nix:152]:
```suggestion
      --replace-fail 'default-version = "0.0"' 'default-version = "${version}"'
```

nit: be more specific in the replace

**Comment by a-n-n-a-l-e-e** [comment]
@ofborg build python311Packages.qcodes

ofborg will not build for python3Packages, need to specify python311Packages or python312Packages in the commit message.

**Comment by a-n-n-a-l-e-e** [comment]
x64-linux build [failed](https://logs.ofborg.org/?attempt_id=237e6cca-42be-4a1a-98f1-21529787f4eb&key=nixos%2Fnixpkgs.288404) but succeeded locally. flaky test should be disabled and reported upstream

**Comment by eclairevoyant** [comment]
> flaky test should be disabled and reported upstream

As a bunch of tests have this issue (microsoft/Qcodes#5551, microsoft/Qcodes#5418), I don't think it's effective to simply disable one test here.

**Comment by a-n-n-a-l-e-e** [comment]
> > flaky test should be disabled and reported upstream
> 
> As a bunch of tests have this issue ([microsoft/Qcodes#5551](https://github.com/microsoft/Qcodes/issues/5551), [microsoft/Qcodes#5418](https://github.com/microsoft/Qcodes/issues/5418)), I don't think it's effective to simply disable one test here.

ok, then disable multiple tests? disabling flaky tests is discussed in https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md

**Comment by eclairevoyant** [comment]
I know how to disable tests; you don't have to quote the manual at me. My point was, if you saw the linked issues, I have no way of knowing which tests are relevant, as (outside of the error) I see no common thread between them.

IMO It's a waste of time to try disabling them one-by-one - either we disable tests entirely or we find a different approach.

**Comment by a-n-n-a-l-e-e** [comment]
> I know how to disable tests; you don't have to quote the manual at me. My point was, if you saw the linked issues, I have no way of knowing which tests are relevant, as (outside of the error) I see no common thread between them.

I'm not sure we're on the same page here so let's reset. So far ofborg has failed twice in the unit tests for x64 linux while they build locally, the failing tests are:
- https://logs.ofborg.org/?attempt_id=237e6cca-42be-4a1a-98f1-21529787f4eb&key=nixos%2Fnixpkgs.288404
```console
FAILED tests/parameter/test_parameter_ramp.py::test_step_ramp 
```
and then the second time ofborg tried to build a different test fails.
- https://logs.ofborg.org/?key=nixos/nixpkgs.288404&attempt_id=cd0196c7-ff57-4511-b313-45ca7452ec9d
```console
FAILED tests/test_monitor.py::test_setup_teardown - RuntimeError: Failed to start server
FAILED tests/test_monitor.py::test_monitor_replace - RuntimeError: Failed to start server
FAILED tests/test_monitor.py::test_double_join - RuntimeError: Failed to start server
ERROR tests/test_monitor.py::test_connection - RuntimeError: Failed to start server
ERROR tests/test_monitor.py::test_instrument_update - RuntimeError: Failed to start server
ERROR tests/test_monitor.py::test_monitor_root_instr[True] - RuntimeError: Failed to start server
ERROR tests/test_monitor.py::test_monitor_root_instr[False] - RuntimeError: Failed to start server
```

neither of the issues you linked mentioned the above test failures and to me appear to be different failures and do not seem relevant to the failures reported by ofborg.

what i was suggesting was creating an issue(s) upstream with the errors from the tests and then disable them: eg:

```nix
  disabledTestPaths = [
    [...]
    # link to created upstream issue 
    "tests/test_monitor.py"
  ];

  disabledTests = [
    [...]
    # link to created upstream issue
    "test_step_ramp"
  ];
```

which seems to be pretty standard for python packages -- eg, disabling flaky tests, which is why i linked in the manual.

> IMO It's a waste of time to try disabling them one-by-one - either we disable tests entirely or we find a different approach. If you have an idea for the latter I'm open to it, else I might just go with the former.

eh, disabling them one by one and reporting them upstream is really the only way they get fixed and we get reliable builds. however, given the scope of the brokenness -- so far both x64 linux builds failed in different ways -- this is drifting farther outside of the scope of this PR and I think there is an argument to be made that it is fine to merge these changes as is.

**Comment by eclairevoyant** [comment]
The way they've designed some of their tests is sloppy and questionable.
It seems that they are checking the _count_ of log messages and asserting on that.

The actual error for the first test failure was

```
E           assert 2 == 1
E            +  where 2 = len([<LogRecord: qcodes.instrument.visa, 20, /nix/store/mvn3797a834h2cyyhbh6izzsw3qbgn2n-python3.11-qcodes-0.44.1/lib/python3.11/site-packages/qcodes/instrument/visa.py, 30, "Closing VISA handle to %s as there are no non weak references to the instrument.">, <LogRecord: qcodes.parameters.parameter_base, 30, /nix/store/mvn3797a834h2cyyhbh6izzsw3qbgn2n-python3.11-qcodes-0.44.1/lib/python3.11/site-packages/qcodes/parameters/parameter_base.py, 765, "cannot sweep test_step from 40 to 'A' - jumping.">])
E            +    where [<LogRecord: qcodes.instrument.visa, 20, /nix/store/mvn3797a834h2cyyhbh6izzsw3qbgn2n-python3.11-qcodes-0.44.1/lib/python3.11/site-packages/qcodes/instrument/visa.py, 30, "Closing VISA handle to %s as there are no non weak references to the instrument.">, <LogRecord: qcodes.parameters.parameter_base, 30, /nix/store/mvn3797a834h2cyyhbh6izzsw3qbgn2n-python3.11-qcodes-0.44.1/lib/python3.11/site-packages/qcodes/parameters/parameter_base.py, 765, "cannot sweep test_step from 40 to 'A' - jumping.">] = <_pytest.logging.LogCaptureFixture object at 0x7ffde6676e90>.records
```

i.e., the test code expects only one log message, but sees two.

This `Closing VISA handle` message is unexpected from the point of view of the test code, as it's a resource management leak (at least, according to a guess from someone on the first linked upstream issue).

The issues I linked above are similar in that they have the same message about `Closing VISA handle` for presumably the same reason. In the second linked issue's case, the assertion was testing for the content of the message, so they "fixed" the test by checking if the expected message was present anywhere in the logs, instead of testing for absolute equality.

However, these other length-based tests are still going to be flaky, especially if there are race conditions (or whatever other causes) for freeing resources.

Looking into the second borgo failure, it also seems to be some resource management issue:
```
  OSError: [Errno 98] error while attempting to bind on address ('127.0.0.1', 59148): address already in use
```

I'm not a python programmer and know nothing about qcodes, so I don't have a way of reliably identifying which tests are susceptible to resource management issues - which is why for me, I don't think disabling tests one-by-one in this case is going to necessarily solve anything, especially if borgo or hydra chokes later on and I have no way to predict that.

Please note that there are over 2400 tests, and none of the test failures are locally reproducible, which is why I consider trial-and-error to be a waste of time.

**Comment by a-n-n-a-l-e-e** [comment]
thanks. looking at hydra https://hydra.nixos.org/job/nixpkgs/trunk/python311Packages.qcodes.x86_64-linux/all there are more good builds than i expected to see. I don't think that disabling all the unit tests is the proper solution at this point. hydra shows the `test_step_ramp` failing once at the current version.

I'll just merge this and create another PR to disable the tests myself.

**Pull request merged**